### PR TITLE
Defer closing resp.Body after issuing an HTTP request

### PR DESCRIPTION
**Pull request by kargakis**
While wandering around

@liggitt PTAL

### CODE REVIEW

File: pkg/auth/authenticator/password/htpasswd/htpasswd.go [hunk 1]
```diff
@@ -116,82 +115,3 @@ func (a *Authenticator) loadIfNeeded() error {
 	}
 	return nil
 }
-
-func testPassword(password, hash string) (bool, error) {
```
Comment (kargakis) [inline pkg/auth/authenticator/password/htpasswd/htpasswd.go]: Ok, I see you need those tests in the authentication. Will update this with just the body closing

Comment (liggitt) [inline pkg/auth/authenticator/password/htpasswd/htpasswd.go]: Yeah, perhaps a better name for the functions would be good, but you can't move them

**Comment by liggitt** [comment]
Thanks... just scanned and found a couple more places that need fixing:
```
external/github/github.go#GetUserIdentity
dockerregistry/client.go#getImage (needs to defer immediately after checking the Do() error)
```

**Comment by kargakis** [comment]
@liggitt updated those as well

File: pkg/dockerregistry/client.go [hunk 2]
```diff
@@ -341,6 +341,7 @@ func (c *connection) getImage(repo *repository, image, userTag string) (*docker.
 	if err != nil {
 		return nil, convertConnectionError(c.url.String(), fmt.Errorf("error getting json for image %q: %v", image, err))
 	}
+	defer resp.Body.Close()
```
Comment (liggitt) [inline pkg/dockerregistry/client.go]: need to remove the defer below

Comment (kargakis) [inline pkg/dockerregistry/client.go]: missed that, updated

**Comment by liggitt** [comment]
[merge]

**Pull request merged**